documentation/assemblies/upgrading/assembly-upgrade-kafka.adoc

```diff
@@ -5,16 +5,54 @@
 [id='assembly-upgrade-kafka-{context}']
 = Strimzi and Kafka upgrades
 
-Upgrading Strimzi is a two-stage process. To upgrade brokers and clients without downtime, you _must_ complete the upgrade procedures in the following order:
+Upgrading Strimzi is a three-stage process. To upgrade brokers and clients without downtime, you _must_ complete the upgrade procedures in the following order:
 
 . Update your Cluster Operator to the latest Strimzi version.
-** xref:assembly-upgrade-cluster-operator-{context}[]
++
+The approach you take depends on how you xref:cluster-operator-{context}[deployed the Cluster Operator].
++
+* If you deployed the Cluster Operator using the installation YAML files, perform your upgrade by modifying the Operator installation files, as described in xref:assembly-upgrade-cluster-operator-{context}[Upgrading the Cluster Operator]. 
++
+* If you deployed the Cluster Operator from {OperatorHub}, use the Operator Lifecycle Manager (OLM) to change the update channel for the Strimzi Operators to the new Strimzi version.
++
+Depending on your chosen upgrade strategy, after updating the channel, either:
++
+** An automatic upgrade is initiated
++
+** A manual upgrade will require approval before the installation begins
++
+For more information on using {OperatorHub} to upgrade Operators, see the {OLMOperatorDocs}.
 
 . Upgrade all Kafka brokers and client applications to the latest Kafka version.
-** xref:assembly-upgrading-kafka-versions-{context}[]
++
+* xref:assembly-upgrading-kafka-versions-{context}[]
+* xref:con-strategies-for-upgrading-clients-{context}[] 
+
+. If applicable, update existing resources to handle deprecated custom resource properties.
++
+* xref:assembly-upgrade-resources-{context}[Strimzi resource upgrades]
```

Comment:
I think having these two points as a separate stages is ok. But they are interchangeable ... you can also first update the resources and then upgrade Kafka. Maybe we should make it more clear?

Reply:
Done.